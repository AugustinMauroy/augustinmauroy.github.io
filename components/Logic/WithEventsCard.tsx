'use client';
import { useTranslations } from 'next-intl';
import type { FC } from 'react';
import EventsCard from '~/components/Common/EnventsCard/index.tsx';
import { isInRange, isToday } from '~/utils/date.ts';

type Event = {
  key: string;
  date?: string;
  startDate?: string;
  endDate?: string;
};

const EVENTS: Event[] = [
  {
    endDate: '2023-06-30',
    key: 'prideMonth',
    startDate: '2023-06-01',
  },
  {
    date: '2023-09-23',
    key: 'bisexualityDay',
  },
  {
    date: '2023-12-01',
    key: 'aidsDay',
  },
];

const WithEventsCard: FC = () => {
  const t = useTranslations('events');
  const currentDate = new Date().toISOString();
  const currentEvents = EVENTS.filter(({ date, startDate, endDate }) => {
    if (date) return isToday(date);
    else if (startDate && endDate)
      return isInRange(currentDate, startDate, endDate);

    return false;
  });

  if (currentEvents.length === 0) return null;

  return (
    <EventsCard
      description={t(`${currentEvents[0]?.key}.description`) || ''}
      title={t(`${currentEvents[0]?.key}.title`) || ''}
    />
  );
};

export default WithEventsCard;
