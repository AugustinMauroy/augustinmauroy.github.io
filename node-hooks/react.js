import { register } from 'node:module';
import { pathToFileURL } from 'node:url';
import jsdom from 'jsdom';

register('nodejs-loaders/testing/css-module', pathToFileURL('./'));

// We doesn't need to load any html/css because it's handle by testing-library
const { document } = new jsdom.JSDOM().window;

/**
 * Make the JSDOM global objects available to the Node.js environment.
 */
global.document = document;
global.window = document.defaultView;
