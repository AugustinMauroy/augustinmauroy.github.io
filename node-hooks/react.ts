// @ts-ignore not backported to 22 and we are on 23
import { register, registerHooks } from 'node:module';
import 'global-jsdom/register';

registerHooks('nodejs-loaders/dev/alias', import.meta.url);
register('@nodejs-loaders/tsx', import.meta.url);
register('@nodejs-loaders/css-module', import.meta.url);
