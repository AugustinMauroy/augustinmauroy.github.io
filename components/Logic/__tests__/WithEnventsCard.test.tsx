import { render } from '@testing-library/react';
import { IntlProvider } from 'next-intl';
import { vi, describe, it, expect } from 'vitest';
import messages from '~/i18n/locales/en.json';
import WithEventsCard from '../WithEnventsCard';

// Mocking the date utility functions
vi.mock('~/utils/date', () => ({
  isToday: (date: string) => date === '2023-09-23',
  isInRange: (currentDate: string, startDate: string, endDate: string) =>
    currentDate >= startDate && currentDate <= endDate,
}));

describe('WithEventsCard', () => {
  it('renders correctly with no events', () => {
    const { container } = render(
      <IntlProvider locale="en" messages={messages}>
        <WithEventsCard />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with a single event', () => {
    const { container } = render(
      <IntlProvider locale="en" messages={messages}>
        <WithEventsCard />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with multiple events', () => {
    const { container } = render(
      <IntlProvider locale="en" messages={messages}>
        <WithEventsCard />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
