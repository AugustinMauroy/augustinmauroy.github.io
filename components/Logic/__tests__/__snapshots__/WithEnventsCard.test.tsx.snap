// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WithEventsCard > renders correctly with a single event 1`] = `
<div>
  <div
    class="_card_2ddc65"
  >
    <svg
      aria-hidden="true"
      class="_shapeTop_2ddc65"
      fill="none"
      height="454"
      viewBox="0 0 454 454"
      width="454"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M453.2 85.9008C406.2 85.9008 368.1 47.8008 368.1 0.800781H340.1H312.1C312.1 47.8008 274 85.9008 227 85.9008C180 85.9008 141.9 47.8008 141.9 0.800781H113.9H85.8998C85.8998 47.8008 47.7998 85.9008 0.799805 85.9008V113.901V141.901C47.7998 141.901 85.8998 180.001 85.8998 227.001C85.8998 274.001 47.7998 312.101 0.799805 312.101V340.101V368.101C47.7998 368.101 85.8998 406.201 85.8998 453.201H113.9H141.9C141.9 406.201 180 368.101 227 368.101C274 368.101 312.1 406.201 312.1 453.201H340.1H368.1C368.1 406.201 406.2 368.101 453.2 368.101V340.101V312.101C406.2 312.101 368.1 274.001 368.1 227.001C368.1 180.001 406.2 141.901 453.2 141.901V113.901V85.9008ZM312.1 227.001C312.1 274.001 274 312.101 227 312.101C180 312.101 141.9 274.001 141.9 227.001C141.9 180.001 180 141.901 227 141.901C274 141.901 312.1 180.001 312.1 227.001Z"
        fill="currentColor"
      />
    </svg>
    <h1>
      Bisexuality Day
    </h1>
    <p>
      Celebrate Bisexuality Day on September 23rd.
    </p>
    <svg
      aria-hidden="true"
      class="_shapeBottom_2ddc65"
      fill="none"
      height="444"
      viewBox="0 0 444 444"
      width="444"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M443.7 222.001C379.9 199.301 349.7 126.401 378.8 65.2008C317.6 94.3008 244.7 64.1008 222 0.300781C199.3 64.1008 126.4 94.3008 65.1998 65.2008C94.2998 126.401 64.0998 199.301 0.299805 222.001C64.0998 244.701 94.2998 317.601 65.1998 378.801C126.4 349.701 199.3 379.901 222 443.701C244.7 379.901 317.6 349.701 378.8 378.801C349.7 317.601 379.9 244.701 443.7 222.001Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`WithEventsCard > renders correctly with multiple events 1`] = `
<div>
  <div
    class="_card_2ddc65"
  >
    <svg
      aria-hidden="true"
      class="_shapeTop_2ddc65"
      fill="none"
      height="454"
      viewBox="0 0 454 454"
      width="454"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M453.2 85.9008C406.2 85.9008 368.1 47.8008 368.1 0.800781H340.1H312.1C312.1 47.8008 274 85.9008 227 85.9008C180 85.9008 141.9 47.8008 141.9 0.800781H113.9H85.8998C85.8998 47.8008 47.7998 85.9008 0.799805 85.9008V113.901V141.901C47.7998 141.901 85.8998 180.001 85.8998 227.001C85.8998 274.001 47.7998 312.101 0.799805 312.101V340.101V368.101C47.7998 368.101 85.8998 406.201 85.8998 453.201H113.9H141.9C141.9 406.201 180 368.101 227 368.101C274 368.101 312.1 406.201 312.1 453.201H340.1H368.1C368.1 406.201 406.2 368.101 453.2 368.101V340.101V312.101C406.2 312.101 368.1 274.001 368.1 227.001C368.1 180.001 406.2 141.901 453.2 141.901V113.901V85.9008ZM312.1 227.001C312.1 274.001 274 312.101 227 312.101C180 312.101 141.9 274.001 141.9 227.001C141.9 180.001 180 141.901 227 141.901C274 141.901 312.1 180.001 312.1 227.001Z"
        fill="currentColor"
      />
    </svg>
    <h1>
      Bisexuality Day
    </h1>
    <p>
      Celebrate Bisexuality Day on September 23rd.
    </p>
    <svg
      aria-hidden="true"
      class="_shapeBottom_2ddc65"
      fill="none"
      height="444"
      viewBox="0 0 444 444"
      width="444"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M443.7 222.001C379.9 199.301 349.7 126.401 378.8 65.2008C317.6 94.3008 244.7 64.1008 222 0.300781C199.3 64.1008 126.4 94.3008 65.1998 65.2008C94.2998 126.401 64.0998 199.301 0.299805 222.001C64.0998 244.701 94.2998 317.601 65.1998 378.801C126.4 349.701 199.3 379.901 222 443.701C244.7 379.901 317.6 349.701 378.8 378.801C349.7 317.601 379.9 244.701 443.7 222.001Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`WithEventsCard > renders correctly with no events 1`] = `
<div>
  <div
    class="_card_2ddc65"
  >
    <svg
      aria-hidden="true"
      class="_shapeTop_2ddc65"
      fill="none"
      height="454"
      viewBox="0 0 454 454"
      width="454"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M453.2 85.9008C406.2 85.9008 368.1 47.8008 368.1 0.800781H340.1H312.1C312.1 47.8008 274 85.9008 227 85.9008C180 85.9008 141.9 47.8008 141.9 0.800781H113.9H85.8998C85.8998 47.8008 47.7998 85.9008 0.799805 85.9008V113.901V141.901C47.7998 141.901 85.8998 180.001 85.8998 227.001C85.8998 274.001 47.7998 312.101 0.799805 312.101V340.101V368.101C47.7998 368.101 85.8998 406.201 85.8998 453.201H113.9H141.9C141.9 406.201 180 368.101 227 368.101C274 368.101 312.1 406.201 312.1 453.201H340.1H368.1C368.1 406.201 406.2 368.101 453.2 368.101V340.101V312.101C406.2 312.101 368.1 274.001 368.1 227.001C368.1 180.001 406.2 141.901 453.2 141.901V113.901V85.9008ZM312.1 227.001C312.1 274.001 274 312.101 227 312.101C180 312.101 141.9 274.001 141.9 227.001C141.9 180.001 180 141.901 227 141.901C274 141.901 312.1 180.001 312.1 227.001Z"
        fill="currentColor"
      />
    </svg>
    <h1>
      Bisexuality Day
    </h1>
    <p>
      Celebrate Bisexuality Day on September 23rd.
    </p>
    <svg
      aria-hidden="true"
      class="_shapeBottom_2ddc65"
      fill="none"
      height="444"
      viewBox="0 0 444 444"
      width="444"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M443.7 222.001C379.9 199.301 349.7 126.401 378.8 65.2008C317.6 94.3008 244.7 64.1008 222 0.300781C199.3 64.1008 126.4 94.3008 65.1998 65.2008C94.2998 126.401 64.0998 199.301 0.299805 222.001C64.0998 244.701 94.2998 317.601 65.1998 378.801C126.4 349.701 199.3 379.901 222 443.701C244.7 379.901 317.6 349.701 378.8 378.801C349.7 317.601 379.9 244.701 443.7 222.001Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
